src/fileimpl.cpp

```diff
@@ -551,27 +551,40 @@ class Grouping
     struct zim_MD5_CTX md5ctx;
     zim_MD5Init(&md5ctx);
 
+    unsigned char ch[CHUNK_SIZE];
     offset_type checksumPos = header.getChecksumPos();
-    offset_type currentPos = 0;
+    offset_type toRead = checksumPos;
+    
     for(auto part = zimFile->begin();
         part != zimFile->end();
         part++) {
       std::ifstream stream(part->second->filename(), std::ios_base::in|std::ios_base::binary);
 
-      char ch;
-      for(/*NOTHING*/ ; currentPos < checksumPos && stream.get(ch).good(); currentPos++) {
-        zim_MD5Update(&md5ctx, reinterpret_cast<const uint8_t*>(&ch), 1);
+      while(toRead>=CHUNK_SIZE && stream.read(reinterpret_cast<char*>(ch),CHUNK_SIZE).good()) {
+        zim_MD5Update(&md5ctx, ch, CHUNK_SIZE);
+        toRead-=stream.gcount();
       }
+
+      // It reads the remaining amount of file when we reach the end of the file
```

Comment:
// It reads the remaining amount of ~~file~~**part** when we reach the end of the ~~file~~**part**

Reply:
When `stream.good()` is true after the while loop, it means that the first condition `toRead>=CHUNK_SIZE `in the loop was triggered. From this we can infer that, we have reached near the end of the file.
I think it should be actually:
`// It reads the remaining amount of part when we reach the end of the file`